core/src/main/java/de/danoeh/antennapod/core/service/BasicAuthorizationInterceptor.java

```diff
@@ -50,6 +53,12 @@ public Response intercept(Chain chain) throws IOException {
         }
 
         Request.Builder newRequest = request.newBuilder();
+        if (0 != StringUtils.compare(
+                response.request().url().toString(),
+                request.url().toString())) {
```

Comment:
Why not `TextUtils.equals()`?

Reply:
good suggestion

Reply:
> Did you test whether the redirect interceptor for password protected feeds still works? E.g., do password protected feeds still store the new redirected URL in the database?

Yes it does follow the redirect and store the new url